bigchaindb/tendermint/lib.py

```diff
@@ -65,23 +65,31 @@ def _process_post_response(self, response, mode):
         if response.get('error') is not None:
             return (500, 'Internal error')
 
+        result = response['result']
+        if mode == MODE_LIST[2]:
+            return self._process_commit_mode_response(result)
+        else:
+            status_code = result['code']
+            return self._process_status_code(status_code,
+                                             'Error while processing transaction')
+
+    def _process_commit_mode_response(self, result):
+        """The implementation below is dumb but clear"""
+        try:
+            check_tx_status_code = result['check_tx']['code']
+            if check_tx_status_code > 0:
+                return (500, 'Error while validating the transaction')
+        except KeyError:
+            pass
+
+        try:
+            deliver_tx_status_code = result['deliver_tx']['code']
+            if deliver_tx_status_code > 0:
+                return (500, 'Error while commiting the transaction')
+        except KeyError:
+            pass
```

Comment:
In case both `try ... except` capture the `KeyError` exception, we return a `202`, am I correct? If so, we should return an error.